code/modules/mob/living/carbon/carbon_defines.dm

```diff
@@ -124,3 +124,9 @@
 	var/bodyshape = BODYSHAPE_HUMANOID
 
 	COOLDOWN_DECLARE(bleeding_message_cd)
+
+/mob/living/carbon/can_perform_action(atom/movable/target, action_bitflags)
+	if(action_bitflags & NEED_VITALITY && (stat == DEAD || stat != CONSCIOUS))
+		to_chat(src, span_warning("You are in no physical condition to do this!"))
+		return FALSE
+	return ..()
```

Comment:
Why's this on the `carbon` level? Stat is on all mobs.

Reply:
Because this check fails for silicons. They have their `stat` set to `NONE` so the condition `stat != CONSCIOUS` always passes which isn't useful for us

Reply:
wtf? `NONE` isn't even a stat. where is this?

Reply:
Silicons get set to `DEAD` when relevant so it should still be fine anyways

Reply:
I believe your right. Let me double check

Reply:
Yup you right. So we still need this in the carbon level because this check will always fail for silicons cause `stat == DEAD` will be true.

Reply:
I am not sure what you mean here, we do typical stat checks for silicons all over the place, they don't *not* use stat. 

They are conscious when idle and dead when dead

Reply:
Let's say i want to change the drop point for autolathe printing & say hypothetically we use `can_perform_action()`.

Now for carbon mobs this proc will succeed because their `stat` variable is set to `CONSCIOUS` & other terms

But for silicon's since their `stat` is always `DEAD` this proc will return false & so they won't be allowed to change the drop point.

In other words we don't want check for a variable that is never used leading t false positives

Reply:
Well shit i must have misunderstood my own code somewhere during testing. Yup you right, I'll move it up